src/components/Dependencies/Dependencies.tsx

```diff
@@ -1,117 +1,77 @@
+import { useHistory } from 'react-router-dom';
 import CardContent from '@material-ui/core/CardContent';
 
-import { DetailContextConsumer, VersionPageConsumerProps } from '../../pages/Version';
-
-import { CardWrap, Heading, Tags, Tag } from './styles';
+import { PackageDependencies } from '../../../types/packageMeta';
+import { DetailContext } from '../../pages/Version';
 import NoItems from '../NoItems';
+import { CardWrap, Heading, Tags, Tag } from './styles';
 
-type DepDetailProps = {
-  name: string;
-  version: string;
-  onLoading?: () => void;
-} & RouteComponentProps;
-
-interface DepDetailState {
-  name: string;
-  version: string;
+interface DependencyBlockProps {
+  title: string;
+  dependencies: PackageDependencies;
 }
 
-class DepDetail extends Component<DepDetailProps, DepDetailState> {
-  constructor(props: DepDetailProps) {
-    super(props);
-    const { name, version } = this.props;
+const DependencyBlock: React.FC<DependencyBlockProps> = ({ title, dependencies }) => {
+  const { enableLoading } = useContext(DetailContext);
+  const history = useHistory();
 
-    this.state = {
-      name,
-      version,
-    };
-  }
+  const deps = Object.entries(dependencies);
 
-  public render(): ReactElement<HTMLElement> {
-    const { name, version } = this.state;
-    const tagText = `${name}@${version}`;
-    return <Tag className={'dep-tag'} clickable={true} label={tagText} onClick={this.handleOnClick} />;
-  }
+  function handleClick(name: string): void {
+    enableLoading && enableLoading();
 
-  private handleOnClick = () => {
-    const { name } = this.state;
-    const { onLoading, history } = this.props;
-
-    onLoading && onLoading();
     history.push(`/-/web/detail/${name}`);
-  };
-}
-
-const WrapperDependencyDetail = withRouter(DepDetail);
-
-class DependencyBlock extends Component<{ title: string; dependencies: [] }> {
-  public render(): ReactElement<HTMLElement> {
-    const { dependencies, title } = this.props;
-    const deps = Object.entries(dependencies) as [];
-
-    return (
-      <DetailContextConsumer>
-        {({ enableLoading }) => {
-          return (
-            <CardWrap>
-              <CardContent>
-                <Heading variant="subtitle1">{`${title} (${deps.length})`}</Heading>
-                <Tags>{this.renderTags(deps, enableLoading)}</Tags>
-              </CardContent>
-            </CardWrap>
-          );
-        }}
-      </DetailContextConsumer>
-    );
   }
 
-  private renderTags = (deps: [], enableLoading?: () => void) =>
-    deps.map(dep => {
-      const [name, version] = dep as [string, string];
-
-      return <WrapperDependencyDetail key={name} name={name} onLoading={enableLoading} version={version} />;
-    });
+  return (
+    <CardWrap>
+      <CardContent>
+        <Heading variant="subtitle1">{`${title} (${deps.length})`}</Heading>
+        <Tags>
+          {deps.map(([name, version]) => (
+            // eslint-disable-next-line
```

Comment:
This is required because of the `jsx-no-bind` rule which is actually not recommanded (we should discuss it in #168)

Reply:
sure...there are some rules I also would like to discuss about